src/App.tsx

```diff
@@ -15,81 +19,144 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-enum SortType {
-  NONE,
-  ALPHABET,
-  LENGTH,
-}
-
-type ReorderOptions = {
-  sortType: SortType,
+// DON'T save goods to the state
+type State = {
   isReversed: boolean,
+  sortType: SortType,
+  showResetButton: boolean,
 };
 
-// Use this function in the render method to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+    showResetButton: false,
+  };
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  alphSort = (event: React.MouseEvent<HTMLElement>) => {
+    const target = event.target as Element;
 
-  return visibleGoods;
-}
+    this.setState({
+      sortType: SortType.ALPHABET,
+    });
+    if (target.classList.contains('is-light')) {
+      this.setState({
+        showResetButton: true,
+      });
+    }
```

Comment:
let's simplify this logic for reset button displaying, instead of using extra flag `showResetButton` you can check if SortType is selected or not